contracts/contracts/poolBooster/AbstractPoolBoosterFactory.sol

```diff
@@ -50,11 +35,21 @@ contract AbstractPoolBoosterFactory is Governable {
 
     // @param address _oSonic address of the OSonic token
     // @param address _governor address governor
-    constructor(address _oSonic, address _governor) {
+    // @param address _centralRegistry address of the central registry
+    constructor(
+        address _oSonic,
+        address _governor,
+        address _centralRegistry
+    ) {
         require(_oSonic != address(0), "Invalid oSonic address");
         require(_governor != address(0), "Invalid governor address");
+        require(
+            _centralRegistry != address(0),
+            "Invalid central registry address"
+        );
 
         oSonic = _oSonic;
+        centralRegistry = IPoolBoostCentralRegistry(_centralRegistry);
```

Comment:
Just noting that bribeAll() will break if the list gets too large. I think we are okay with this as long as the boosters themselves can have bribe called outside the factory.

Reply:
Yeah good point. We could limit the number of poolBoosters per factory by deploying multiple factories even when the codebase doesn't change. To not hit that gas usage limit.

Or we could pass an offset & limit to the bribeAll() function, but that seems too messy. 

I guess we will need to handle the loss of convenience on the operational side.